File: app/src/main/java/rusha/x/recipe/details/IngredientViewHolder.java
package rusha.x.recipe.details;

import androidx.recyclerview.widget.RecyclerView;

import rusha.x.databinding.RecipeDetailsItemBinding;
import rusha.x.recipe.RecipeIngredient;

/**
 * Держатель отображения ячейки
 */
public class IngredientViewHolder extends RecyclerView.ViewHolder {

    private RecipeDetailsViewModel viewModel;
    private RecipeDetailsItemBinding binding;

    public IngredientViewHolder(RecipeDetailsViewModel viewModel, RecipeDetailsItemBinding binding) {
        super(binding.getRoot());
        this.viewModel = viewModel;
        this.binding = binding;
    }

    /**
     * Держатель ячейки заполняет отображение ячейки данными
     */
    public void bind(RecipeIngredient ingredient) {
        binding.getRoot().setOnClickListener((v) -> {
            viewModel.onIngredientClick(ingredient);
        });
        binding.nameItemView.setText(ingredient.product.name);
        binding.countView.setText(String.valueOf(ingredient.countInRecipe));
    }
}

File: app/src/main/java/rusha/x/recipe/details/IngredientViewHolder.kt
package rusha.x.recipe.details

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.recipe_details_item.view.*
import rusha.x.recipe.Recipe

/**
 * Держатель отображения ячейки
 */
class IngredientViewHolder(
    val viewModel: RecipeDetailsViewModel,
    /**
     * Отображение ячейки (ConstraintLayout из recipe_details_item.xml)
     */
    val cellView: View
) : RecyclerView.ViewHolder(cellView) {

    /**
     * Держатель ячейки заполняет отображение ячейки данными
     */
    fun bind(ingredient: Recipe.Ingredient) {
        cellView.setOnClickListener {
            viewModel.onIngredientClick(ingredient)
        }
        cellView.nameItemView.text = ingredient.product.name
        cellView.countView.text = ingredient.countInRecipe.toString()
    }
}